crates/engine/tree/src/tree/mod.rs

```diff
@@ -2375,11 +2375,27 @@ where
         let (state_root_handle, state_root_task_config, state_hook) =
             if is_descendant_of_persisting_blocks && self.config.use_state_root_task() {
                 let consistent_view = ConsistentDbView::new_with_latest_tip(self.provider.clone())?;
-                let state_root_config = StateRootConfig::new_from_input(
-                    consistent_view.clone(),
-                    self.compute_trie_input(consistent_view, block.header().parent_hash())
-                        .map_err(|e| InsertBlockErrorKind::Other(Box::new(e)))?,
-                );
+
+                // Compute trie input
+                let trie_input_start = Instant::now();
+                let trie_input = self
+                    .compute_trie_input(consistent_view.clone(), block.header().parent_hash())
+                    .map_err(|e| InsertBlockErrorKind::Other(Box::new(e)))?;
+                let trie_input_elapsed = trie_input_start.elapsed();
+                self.metrics
+                    .block_validation
+                    .trie_input_duration
+                    .set(trie_input_elapsed.as_secs_f64());
+
+                // Create state root config
+                let config_start = Instant::now();
```

Comment:
we can safe one now call here if calculate `trie_input_elapsed` as `config_start - trie_input_start`, see elapsed impl